pieces/shared/menu-button/container.tsx

```diff
@@ -0,0 +1,112 @@
+import { ReactNode, RefObject, useCallback, useEffect, useReducer, useRef } from 'react';
+
+import { css } from '@emotion/react';
+import { bind } from 'bind-event-listener';
+
+import Trigger from './trigger';
+import Menu from './menu';
+
+const containerStyles = css({
+  display: 'flex',
+  position: 'relative',
+});
+
+type MenuButtonState =
+  | {
+      isOpen: true;
+      initialFocus: 'first' | 'last';
+    }
+  | {
+      isOpen: false;
+      shouldResetFocus: boolean;
+    };
+
+type MenuButtonAction =
+  | {
+      type: 'open';
+      initialFocus: 'first' | 'last';
+    }
+  | {
+      type: 'close';
+      shouldResetFocus: boolean;
+    };
+
+function reducer(_: MenuButtonState, action: MenuButtonAction): MenuButtonState {
+  if (action.type === 'open') {
+    return {
+      isOpen: true,
+      initialFocus: action.initialFocus,
+    };
+  }
+
+  if (action.type === 'close') {
```

Comment:
You don't need this guard. Typescript whill know that at this point the only action.type available is 'close'. 

So you can remove the guard and the throw below as well

Reply:
fixed :D